File: 04.Selection Sort/selection_sort.c
#include <stdio.h>

void blank_line(void);
void selection_sort(int A[], int array_len);
void show_array(int A[], int len);

int main(void) {

    int Array[] = {23,7,2,8,43,6,78,2,7890,7,235346,23};
    printf("Before Sorting...\n");
    show_array(Array, sizeof(Array)/sizeof(int));
    selection_sort(Array, sizeof(Array)/sizeof(int));
    printf("After Sorting...\n");
    show_array(Array, sizeof(Array)/sizeof(int));
    return 0;
}


void selection_sort(int A[], int array_len) {
    int i, smallest, j, temp;
    for (i = 0; i < array_len-1; i++) {
        smallest = i;
        for (j = i+1; j < array_len; j++) {
            if (A[j] < A[smallest]) {
                smallest = j;
            }
        }
        if (smallest != i) {
            temp = A[i];
            A[i] = A[smallest];
            A[smallest] = temp;
        }

    }
}

void show_array(int A[], int len){
    int i;
    for (i = 0; i < len; i++){
        printf("%d ", A[i]);
    }
    blank_line();
}

void blank_line(void){
    printf("\n");
}

File: 04.Selection Sort/selection_sort.java
class selection_sort{
    public static void main(String args[]){
        int array[] = {23,7,2,8,43,6,78,2,7890,7,235346,23};

        System.out.println("Before sorting...");
        ShowArray(array);
        array = selectionSort(array);
        System.out.println("After sorting...");
        ShowArray(array);
    }

    public static int[] selectionSort(int list_of_numbers[]){
        int i, j, smallest, temp;
        for (i=0; i < list_of_numbers.length; i++){
            smallest = i;
            for(j=i+1; j < list_of_numbers.length; j++){
                if (list_of_numbers[j] < list_of_numbers[smallest]){
                    smallest = j;
                }
            }
            if (smallest != i) {
                temp = list_of_numbers[i];
                list_of_numbers[i] = list_of_numbers[smallest];
                list_of_numbers[smallest] = temp;
            }
        }
        return list_of_numbers;
    }

    public static void ShowArray(int arr[]){
        for (int i=0; i < arr.length; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}